extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/DbCheckingPMetaSupplier.java

```diff
@@ -0,0 +1,170 @@
+import com.hazelcast.dataconnection.DataConnectionService;
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.jet.JetException;
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.datamodel.Tuple2;
+import com.hazelcast.jet.impl.util.Util;
+import com.hazelcast.jet.mongodb.dataconnection.MongoDataConnection;
+import com.hazelcast.jet.pipeline.DataConnectionRef;
+import com.hazelcast.spi.impl.NodeEngine;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoDatabase;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.security.Permission;
+import java.util.List;
+import java.util.function.Function;
+
+import static com.hazelcast.jet.datamodel.Tuple2.tuple2;
+import static com.hazelcast.jet.mongodb.impl.MongoUtilities.checkCollectionExists;
+import static com.hazelcast.jet.mongodb.impl.MongoUtilities.checkDatabaseExists;
+
+/**
+ * A {@link ProcessorMetaSupplier} that will check if requested database and collection exist before creating
+ * the processors.
+ */
+public class DbCheckingPMetaSupplier implements ProcessorMetaSupplier {
+
+    private final Permission requiredPermission;
+    private final boolean shouldCheck;
+    private final ProcessorMetaSupplier standardForceOnePMS;
+    private boolean forceTotalParallelismOne;
+    private final String databaseName;
+    private final String collectionName;
+    private final ProcessorSupplier processorSupplier;
+    private final SupplierEx<? extends MongoClient> clientSupplier;
+    private final DataConnectionRef dataConnectionRef;
+    private int preferredLocalParallelism;
+
+    /**
+     * Creates a new instance of this meta supplier.
+     */
+    public DbCheckingPMetaSupplier(@Nullable Permission requiredPermission,
+                                   boolean shouldCheck,
+                                   boolean forceTotalParallelismOne,
+                                   @Nullable String databaseName,
+                                   @Nullable String collectionName,
+                                   @Nullable SupplierEx<? extends MongoClient> clientSupplier,
+                                   @Nullable DataConnectionRef dataConnectionRef,
+                                   @Nonnull ProcessorSupplier processorSupplier,
+                                   int preferredLocalParallelism
+    ) {
+        this.requiredPermission = requiredPermission;
+        this.shouldCheck = shouldCheck;
+        this.forceTotalParallelismOne = forceTotalParallelismOne;
+        this.databaseName = databaseName;
+        this.collectionName = collectionName;
+        this.processorSupplier = processorSupplier;
+        this.clientSupplier = clientSupplier;
+        this.dataConnectionRef = dataConnectionRef;
+        this.preferredLocalParallelism = forceTotalParallelismOne ? 1 : preferredLocalParallelism;
```

Comment:
`forceTotalParallelismOne` can change after constructor

Reply:
but you use the same condition in preferredLocalParallelism - not readable

Reply:
changes setter method as well - c19ee2aba108026791d6b63d7d043d5c7011328d